src/main/java/app/sami/languageWeb/auth/services/AuthenticateUserService.java

```diff
@@ -19,12 +20,12 @@ public class AuthenticateUserService {
 
 
     public AuthenticationResponse authenticate(AuthenticationRequest request){
-        authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(
-                request.getId(),
+        /*authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(
+                request.getEmail(),
                 request.getUserPassword()
-        ));
-        User user = userRepository.findById(request.getId())
-                .orElseThrow(
+        ));*/
+        User user = userRepository.findByEmail(request.getEmail())
+                .orElseThrow(NotFoundException::new
                 );
         String jwtToken = jwtService.generateToken(user);
```

Comment:
^ this shows that you are lacking critical tests here. Something that actually checks the 3 cases:
1. Valid authentication
2. Invalid password
3. Unknown email

If you had those, the tests would fail and you'd know you were missing something